File: go-code/models/order.go
package models

import "math"

type Order struct {
	Name      string
	Amount    int
	ItemPrice float64
	Quantity  int
}

// after the refactor for extract-variable
func (order *Order) Price() float64 {

	return order.basePrice() - order.discount() + order.shippingCost()
}

func (order *Order) shippingCost() float64 {

	return math.Min(order.basePrice()*0.1, 100)
}

func (order *Order) discount() float64 {

	return math.Max(0, float64(order.Quantity-500)) * 0.05
}

func (order *Order) basePrice() float64 {

	return float64(order.Quantity) * order.ItemPrice
}

File: java-code/src/main/java/refactoring/model/Order.java
package refactoring.model;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Order {

  private int amount;
  private int quantity;
  private int itemPrice;

  //was taken to Order class as a result of extract variable refactoring
  public double price() {
    return this.basePrice() - quantityDiscount() + getShipping();
  }

  private double getShipping() {
    return Math.min(this.basePrice() * 0.1, 100);
  }

  private double quantityDiscount() {
    return Math.max(0, this.quantity - 500) * 0.05;
  }

  private int basePrice() {
    return this.quantity * this.itemPrice;
  }
}
